File: tool/uapicodegen/uapicodegen/uapi/internal/generation/GenerateRandomFn.py
from typing import TYPE_CHECKING, cast


if TYPE_CHECKING:
    from ...RandomGenerator import RandomGenerator
    from ...internal.types.UStruct import UStruct
    from ...internal.types.UTypeDeclaration import UTypeDeclaration
    from ...internal.generation.GenerateContext import GenerateContext


def generate_random_fn(blueprint_value: object, use_blueprint_value: bool, call_tags: dict[str, 'UStruct'], ctx: 'GenerateContext') -> object:
    from ...internal.generation.GenerateRandomUnion import generate_random_union

    return generate_random_union(blueprint_value, use_blueprint_value, call_tags, ctx)

File: lib/go/internal/types/GenerateRandomFn.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// GenerateRandomFn delegates to GenerateRandomUnion for function unions.
func GenerateRandomFn(blueprintValue any, useBlueprintValue bool, callTags map[string]*TStruct, ctx *GenerateContext) any {
	return GenerateRandomUnion(blueprintValue, useBlueprintValue, callTags, ctx)
}
